test/src/tests-session-manager.ts

```diff
@@ -0,0 +1,860 @@
+                expect(persistenceSpy.called).to.equal(true);
+            });
+
+            it('should end log StartingEndSession message and return early if Persistence is undefined', () => {
+                mParticle.init(apiKey, window.mParticle.config);
+
+                const mpInstance = mParticle.getInstance();
+                sinon
+                    .stub(mpInstance._Persistence, 'getPersistence')
+                    .returns(null);
+
+                const consoleSpy = sinon.spy(mpInstance.Logger, 'verbose');
+
+                mpInstance._SessionManager.endSession();
+
+                // This path currently goes through the !cookies path,
+                // which doesn't really return anything except for logging the
+                // initial StartingEndSession message
+                expect(consoleSpy.getCall(0).firstArg).to.equal(
+                    Messages.InformationMessages.StartingEndSession
+                );
+
+                // Should only log a single verbose log
+                expect(consoleSpy.getCalls().length).to.equal(1);
+            });
+
+            it('should log a NoSessionToEnd Message if Persistence Exists but does not return an sid', () => {
+                mParticle.init(apiKey, window.mParticle.config);
+
+                const mpInstance = mParticle.getInstance();
+                sinon
+                    .stub(mpInstance._Persistence, 'getPersistence')
+                    .returns({ gs: {} });
+
+                const consoleSpy = sinon.spy(mpInstance.Logger, 'verbose');
+
+                mpInstance._SessionManager.endSession();
+
+                // Should log initial StartingEndSession and NoSessionToEnd messagesk
+                expect(consoleSpy.getCalls().length).to.equal(2);
+                expect(consoleSpy.lastCall.firstArg).to.equal(
+                    Messages.InformationMessages.NoSessionToEnd
+                );
+            });
+
+            it('should log an AbandonedEndSession message if unable to log', () => {
+                mParticle.init(apiKey, window.mParticle.config);
+
+                const mpInstance = mParticle.getInstance();
+                sinon
+                    .stub(mpInstance._Persistence, 'getPersistence')
+                    .returns({ gs: {} });
+
+                sinon.stub(mpInstance._Helpers, 'canLog').returns(false);
+
+                const consoleSpy = sinon.spy(mpInstance.Logger, 'verbose');
+
+                mpInstance._SessionManager.endSession();
+
+                // Should log initial StartingEndSession and AbandonEndSession messagesk
+                expect(consoleSpy.getCalls().length).to.equal(2);
+                expect(consoleSpy.lastCall.firstArg).to.equal(
+                    Messages.InformationMessages.AbandonEndSession
+                );
+            });
+
+            it('should log an AbandonedEndSession message if Store is not enabled', () => {
```

Comment:
Wording is a bit confusing here. it's not that Store is or isn't enabled, it's that the flag isEnabled is `false`.  It looks like isEnabled is only set when `optOut` is called. The flag itself is definitely not well named, and maybe we should address that in a future refactor to be more clear as to what is going on (is SDK logging enabled or not).

```suggestion
            it('should log an AbandonedEndSession message if Store.isEnabled is false', () => {
```